src/algebra/poly/authenticated_poly.rs

```diff
@@ -379,6 +476,65 @@ impl<C: CurveGroup> Div<&DensePolynomialResult<C>> for &AuthenticatedDensePoly<C
 }
 impl_borrow_variants!(AuthenticatedDensePoly<C>, Div, div, /, DensePolynomialResult<C>, C: CurveGroup);
 
+/// Authenticated division, i.e. division in which the divisor is a secret shared polynomial
+///
+/// We follow the approach of: https://iacr.org/archive/pkc2006/39580045/39580045.pdf (Section 5)
+///
+/// To see why this method holds, consider the `rev` operation for a polynomial a(x):
+///     rev(a) = x^deg(a) * a(1/x)
+/// Note that this operation is equivalent to reversing the coefficients of a(x)
+/// For f(x) / g(x) where deg(f) = n, deg(g) = m, the objective of a division with
+/// remainder algorithm is to solve:
+///     f(x) = g(x)q(x) + r(x)
+/// for q(x), r(x) uniquely where deg(r) < deg(g).
+///
+/// We could solve for q(x) easily if:
+///     1. We could "mod out" r(x) and
+///     2. If g^{-1}(x) exists
+/// The rev operator provides a transformation that makes both of these true:
+///     rev(f) = rev(g) * rev(q) + x^{n - m + 1} * rev(r)
```

Comment:
the way you defined `rev` at the beginning of this doc comment as "knowing" `deg(a)`, as opposed to the paper's definition `rev_k(a) = x^k * a(1/x)`, makes intuitive sense, but makes the equation above correct only if `deg(r) = m-1`.

we could do the following:
```suggestion
///     rev(f) = rev(g) * rev(q) + x^{n - m + 1} * x^{m-1} * r(1/x)
```

but that doesn't read as nicely, so I'd actually be in favor of my suggestion above on lines 484-485

Reply:
we only ever instantiate it with the degree, no point in adding complexity to an already obscure derivation